### DISCUSSION TITLE
Message qt.qml.method.behavior...

### DISCUSSION THREAD

**Discussion by octaeder** [texstudio-org/texstudio#3547] [Q&A] [answered]
I recently noticed a debug message shown below in current dev (7.4.3) and 4.7.1 (msys2, Win10, Qt6). As an example it appears when using `ctrl+shift+i` (insert \item):

``
qt.qml.method.behavior: :9: Calling C++ methods with 'this' objects different from the one they were retrieved from is broken, due to historical reasons. The original object is used as 'this' object. You can allow the given 'this' object to be used by setting 'pragma NativeMethodBehavior: AcceptThisObject'
``

Call chain is: insertFromAction -->execMacro --> runScript --> run --> QJSValue result=engine->evaluate(m_script);

The evaluate method throws the message. Is there something we can/should do?

**Comment by sunderme** [comment, marked as answer]
it comes from 
scriptengine.cpp
```
editorValue.setProperty("insertSnippet", scriptJS.property("insertSnippet"));
```

Qt does not like that a special function is added to the javascript editor object.
I have not seen a way to silence the warning, I don't see adverse effects, so I will ignore the warning.
